hoomd/md/pytest/test_alj.py

```diff
@@ -0,0 +1,246 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+import numpy as np
+import pytest
+
+import hoomd
+import hoomd.conftest
+from hoomd import md
+
+
+class ConservationDataStorer(hoomd.custom.Action):
+    """Store energy and velocities from a simulation."""
+
+    def __init__(self, sim, thermo):
+        self._sim = sim
+        self._thermo = thermo
+        self._velocities = []
+        self._energies = []
+
+    def act(self, timestep):
+        with self._sim.state.cpu_local_snapshot as snap:
+            self._energies.append(self._thermo.potential_energy
+                                  + self._thermo.kinetic_energy)
+            self._velocities.append(np.array(snap.particles.velocity,
```

Comment:
Momentum in the local snapshot will not be conserved as particles in the local domain interact with particles in neighboring domains. Compute the momentum for the total system.

`Integrator::computeTotalMomentum` does this, but it is not currently exposed to Python. I will add in another PR.